### DISCUSSION TITLE
ExternalScriptsHandle - how to block rendering until external script loaded and run?

### DISCUSSION THREAD

**Discussion by tomer-yechiel** [sergiodxa/remix-utils#287] [Q&A] [answered]
Hi,
I put ExternalScriptsHandle in nested route. if i do refresh when i am on the nested route everything is ok and the component render after the script is loaded.
if i refresh from another route and then navigate to the nested route, the component is render before the script loaded.
i configured it with async/defer = false.

**Comment by sergiodxa** [comment, marked as answer]
There’s no way, Remix doesn’t give me a way to block the navigation until another thing happens.

**Comment by tomer-yechiel** [comment]
for anyone who encounter this issue, just put the external script at the root.tsx

**Reply by sergiodxa** [reply]
If you mean the ExternalScripts component that should always go in `root`, the same way Links and Meta goes in root.
